lib/transformSync.js

```diff
@@ -14,7 +14,7 @@ exports.transform = function(source, options) {
 			generatedFile: path.replace(/\._coffee$/, '.coffee'),
 			sourceFiles: [path],
 		});
-		if (options.ignore && options.ignore(path)) {
+		if ((options.ignore && options.ignore(path)) || /\.coffee$/.test(options.ext || path)) {
```

Comment:
Not sure if you can use ES6 in this file, but there is [`String::endsWith`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) now :smile: